tools/manifest/sourcefile.py

```diff
@@ -242,9 +242,7 @@ def __getstate__(self):
 
         if "__cached_properties__" in rv:
             cached_properties = rv["__cached_properties__"]
-            for key in rv.keys():
-                if key in cached_properties:
-                    del rv[key]
+            rv = {key:value for key, value in iteritems(rv) if key not in cached_properties}
```

Comment:
Why this change?

Reply:
Seems unrelated indeed, but python3 complains because we're indeed changing a dictionary while iterating over it, that's why I decided to use comprehension here.

Reply:
Oh right, yes.

I wonder if this is performance-critical enough to write the whole function as:

```
if "__cached_properties__" in self.__dict__:
    cached_properties = self.__dict__["__cached_properties__"]
    rv = {key:value for key, value in iteritems(self.__dict__) if key not in cached_properties}
else:
    rv = self.dict.copy()
return rv
```
Maybe not, since we just spend so much time parsing HTML that a dict copy doesn't matter. But perhaps it would be worth checking for a performance regression when we rebuild the manifest with this branch?

Reply:
I'll get some numbers

Reply:
Did you try measuring this? I don't expect it to be a big problem, but I'm at least interested.

Reply:
Oh I forgot to reply to this. Yes, I did take measurements. I got the following after running the `test_wpt.py` tests before & after. 

average duration (in seconds)
before: 1,98 e-06  
after:    5,61 e-06

There are indeed differences but they're so small that I think we can live with that. As an example I got 74k meassurements running those tests. If I sum all the durations I get 0,4s after and 0,15s before, so less than 0,3s of difference running all the `test_wpt.py` tests.

Reply:
Interesting. I'm happy for this to land now, but maybe file an issue (or just make a followup PR) to take that change. Saving a third of a second on a full rebuild isn't much, but it is within the realm of small improvements that are worthwhile when you're trying to optimise.

Reply:
OK, I'll do